patches/tModLoader/Terraria/ModLoader/UI/ModBrowser/UIModDownloadItem.cs

```diff
@@ -81,7 +81,7 @@ internal class UIModDownloadItem : UIPanel
 			_moreInfoButton.OnClick += ViewModInfo;
 			Append(_moreInfoButton);
 
-			if (!ModLoader.versionedName.Contains(ModDownload.ModloaderVersion)) {
+			if ((BuildInfo.IsStable || BuildInfo.IsPreview) && !ModLoader.versionedName.Contains(ModDownload.ModloaderVersion)) {
```

Comment:
`!BuildInfo.IsDev`
But also, this is gonna break..
I see below you have `// Use the stable version of the mod for publishing metadata, not the preview version!`

1. Old mod versions should be assumed forward compatible
2. If a mod is new and only published on preview... should there be an appropriate message?